cut_flowline_module.py

```diff
@@ -0,0 +1,136 @@
+            crs (QgsCoordinateReferenceSystem): Coordinate reference system for the new layer
+        """
+        cut_layer = QgsVectorLayer(f"Point?crs={crs.authid()}", "CutPoints", "memory")
+        provider = cut_layer.dataProvider()
+
+        # Add fields: 'id' (Int), 'tag' (String)
+        provider.addAttributes([
+            QgsField("id", QVariant.Int),
+            QgsField("tag", QVariant.String)
+        ])
+        cut_layer.updateFields()
+
+        # Create new features with sequential IDs and tag = "fly-over"
+        new_features = []
+        for i, original_feature in enumerate(cut_features, start=1):
+            new_feature = QgsFeature()
+            new_feature.setGeometry(original_feature.geometry())
+            # 'id' -> i; 'tag' -> "fly-over"
+            new_feature.setAttributes([i, "fly-over"])
+            new_features.append(new_feature)
+
+        # Add new features to the layer
+        provider.addFeatures(new_features)
+        QgsProject.instance().addMapLayer(cut_layer)
+        self.iface.messageBar().pushMessage(
+            "Info",
+            "Cut points created with sequential IDs and tag = 'fly-over'.",
+            level=Qgis.Info
+        )
+
+    def select_points(self):
+        """
+        Select two points from the point layer.
+
+        Returns:
+            tuple: A tuple containing the two point IDs (smaller_id, larger_id),
+                  or None if selection is invalid
+        """
+        selected_layer = self.layer_utils.get_valid_selected_layer(
+            [QgsWkbTypes.GeometryType.PointGeometry]
+        )
+        if selected_layer is None:
+            return None
+
+        features = list(selected_layer.getSelectedFeatures())
+
+        if len(features) != 2:
+            self.iface.messageBar().pushMessage(
+                "Error",
+                "Please select exactly two points.",
+                level=Qgis.Warning
+            )
+            return None
+
+        point1_id = features[0]["id"]
+        point2_id = features[1]["id"]
+
+        # Return IDs in order (smaller, larger)
+        return (min(point1_id, point2_id), max(point1_id, point2_id))
+
+    def cut_action(self):
+        """Execute the cut action on the selected points."""
+        selected_ids = self.select_points()
+        if selected_ids is None:
+            return
+
+        min_id, max_id = selected_ids
+        point_layer = self.layer_utils.get_valid_selected_layer(
+            [QgsWkbTypes.GeometryType.PointGeometry]
+        )
+
+        if point_layer is None:
+            self.iface.messageBar().pushMessage(
+                "Error",
+                "Please select a valid point layer.",
+                level=Qgis.Warning
+            )
+            return
+
+        # Create a list of features within the ID range
+        cut_features = []
+        for feature in point_layer.getFeatures():
+            point_id = feature["id"]
```

Comment:
What if the feature doesn't have an `id` field?
Example solution: 
```suggestion
            if selected_layer.fields().indexFromName(self.qgis_field_name) == -1:
                self.iface.messageBar().pushMessage(
                    "Please select a vector layer with an id field",
                    level=Qgis.Warning,
                    duration=4,
                )
                return            
            point_id = feature["id"]
```

Reply:
this still needs a fix. How can sorted features be extracted?